src/GameLogic/GameSession.ts

```diff
@@ -21,10 +23,8 @@ export class GameSession {
 
 	private readonly _recordings: ActionSequence[];
 
-	private readonly _levelFactory: () => Level;
-
-	constructor(levelFactory: () => Level, config: GameSessionConfig = {}) {
-		this._levelFactory =  levelFactory;
+	constructor(levelBlueprint: Level, config: GameSessionConfig = {}) {
+		this.levelBlueprint = levelBlueprint.clone();
```

Comment:
Hm. I am tempted to say you don't need to make a clone here, because the chances of someone passing a level here and then changing it are extremely slim. But on the other hand making the clone here doesn't do anything bad except wasting some resources. Your call I guess :).

Reply:
I can see the arguments for either way. I think I'll remove it for now and we can add it back in if we want to manipulate the blueprint for any reason.